elastic/kibana
# [Discover] Update data view id on adhoc data view change

## Summary

Closes #141908

With this, now adhoc data view id will be updated of on each change from Discover page. 

### Test
- Create adhoc data view in Discover
- Save search
- Change data view by adding new field or editing one (`id` in url should be changed)
- Navigate to Lens using `Visualize` button
- Save object in library
- Inspect both saved objects and check that data view ids are different

This PR also adds warning toast which notifies about that data view id and some of the filters `index` property are different. It might happend sometimes when going back. See video below.

https://user-images.githubusercontent.com/39378793/194045201-8e0172ad-072f-47a7-b9a4-ea15084bc737.MP4





### Checklist
